src/aggregate.jl

```diff
@@ -1,317 +1,384 @@
-    return reference_map.varmap[vref]
+function Base.getindex(ref_map::GraphReferenceMap, vref::NodeVariableRef)
+    return ref_map.var_map[vref]
 end
 
-function Base.getindex(reference_map::AggregateMap, cref::JuMP.ConstraintRef)
-    return reference_map.conmap[cref]
+function Base.getindex(ref_map::GraphReferenceMap, cref::JuMP.ConstraintRef)
+    return ref_map.con_map[cref]
 end
 
-#NOTE: Quick fix for aggregating object dictionaries
-function Base.getindex(reference_map::AggregateMap, value::Any)
+# NOTE: Quick fix for aggregating object dictionaries
+function Base.getindex(ref_map::GraphReferenceMap, value::Any)
     return value
 end
 
-Base.broadcastable(reference_map::AggregateMap) = Ref(reference_map)
+Base.broadcastable(ref_map::GraphReferenceMap) = Ref(ref_map)
 
 function Base.setindex!(
-    reference_map::AggregateMap,
+    ref_map::GraphReferenceMap,
     graph_cref::JuMP.ConstraintRef,
     node_cref::JuMP.ConstraintRef,
 )
-    return reference_map.conmap[node_cref] = graph_cref
+    return ref_map.con_map[node_cref] = graph_cref
 end
 
 function Base.setindex!(
-    reference_map::AggregateMap, graph_vref::JuMP.VariableRef, node_vref::JuMP.VariableRef
+    ref_map::GraphReferenceMap, graph_vref::NodeVariableRef, node_vref::NodeVariableRef
 )
-    return reference_map.varmap[node_vref] = graph_vref
+    return ref_map.var_map[node_vref] = graph_vref
 end
 
-# AggregateMap(node::OptiNode) = AggregateMap(node,Dict{JuMP.VariableRef,JuMP.VariableRef}(),Dict{JuMP.ConstraintRef,JuMP.ConstraintRef}(),Dict{LinkConstraintRef,JuMP.ConstraintRef}())
-
-function AggregateMap()
-    return AggregateMap(
-        Dict{JuMP.VariableRef,JuMP.VariableRef}(),
-        Dict{JuMP.ConstraintRef,JuMP.ConstraintRef}(),
-        Dict{LinkConstraintRef,JuMP.ConstraintRef}(),
+function GraphReferenceMap()
+    return GraphReferenceMap(
+        OrderedDict{NodeVariableRef,NodeVariableRef}(),
+        OrderedDict{JuMP.ConstraintRef,JuMP.ConstraintRef}(),
     )
 end
-function Base.merge!(ref_map1::AggregateMap, ref_map2::AggregateMap)
-    merge!(ref_map1.varmap, ref_map2.varmap)
-    return merge!(ref_map1.conmap, ref_map2.conmap)
+function Base.merge!(ref_map1::GraphReferenceMap, ref_map2::GraphReferenceMap)
+    merge!(ref_map1.var_map, ref_map2.var_map)
+    return merge!(ref_map1.con_map, ref_map2.con_map)
 end
 
-#############################################################################################
-# Aggregate Functions
-#############################################################################################
+### Aggregate Functions
+
 """
-    aggregate(graph::OptiGraph)
+    aggregate(graph::OptiGraph; return_mapping=true)
 
-Aggregate the optigraph `graph` into a new optinode.  Return an optinode and a dictionary which maps optinode variable and
-constraint references to the original optigraph.
+Aggregate an optigraph into a graph containing a single optinode.
+"""
+function aggregate(graph::OptiGraph; name=gensym())
```

Comment:
The doc string here includes a `return_mapping`  key word argument, but that doesn't look like it's included in the function. Also, the function has a `name` key word argument that is not in the doc string.

Reply:
fixed